packages/smooth_app/lib/pages/product/product_page.dart

```diff
@@ -73,13 +78,35 @@ class _ProductPageState extends State<ProductPage> {
     _updateHistory(context);
   }
 
+  Future<void> _refreshIfNeeded(final ProductTimestamp productTimestamp) async {
+    final int? millisecondsSinceEpoch =
+        await productTimestamp.getTimestamp(widget.product);
+    if (millisecondsSinceEpoch == null) {
+      return;
+    }
+    final int nowInMillis = DateTime.now().millisecondsSinceEpoch;
+    final int elapsedInMillis = nowInMillis - millisecondsSinceEpoch;
+    print('last modified was ${elapsedInMillis ~/ 1000} seconds ago');
+    if (_duration.inMilliseconds < elapsedInMillis) {
+      print('should refresh');
+    }
+  }
+
   @override
   Widget build(BuildContext context) {
     final LocalDatabase localDatabase = context.watch<LocalDatabase>();
     final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
     if (_first) {
       _first = false;
       _product = widget.product;
+      _refreshIfNeeded(
```

Comment:
This should probably be done inside a lifecycle method like `didChangeDependencies()`.

Reply:
I didn't know `didChangeDependencies()`.
I'm not sure I understand the notion of "change dependencies"; in my mind I just want the code to be called once, for the first build, and that's it.